assembly-syntax/src/ast/instruction/mod.rs

```diff
@@ -211,9 +212,7 @@ pub enum Instruction {
     PushU32(u32),
     PushFelt(Felt),
     PushWord(WordValue),
-    PushU8List(Vec<u8>),
-    PushU16List(Vec<u16>),
-    PushU32List(Vec<u32>),
+    PushWordSlice(Immediate<IntValue>, Range<usize>),
```

Comment:
I think we should call this just `PushSlice`, as we don't need to necessarily require that the immediate is a word (though in practice that is likely to be the common case).